src/Http/Http.Extensions/src/ProblemDetailsServiceCollectionExtensions.cs

```diff
@@ -40,7 +40,8 @@ public static IServiceCollection AddProblemDetails(this IServiceCollection servi
         // Adding default services;
         services.TryAddSingleton<IProblemDetailsService, ProblemDetailsService>();
         services.TryAddEnumerable(ServiceDescriptor.Singleton<IProblemDetailsWriter, DefaultProblemDetailsWriter>());
-        services.TryAddEnumerable(ServiceDescriptor.Singleton<IPostConfigureOptions<JsonOptions>, ProblemDetailsJsonOptionsSetup>());
+        // Use IConfigureOptions (instead of post-configure) so the registration gets added/invoked relative to when AddProblemDetails() is called.
+        services.TryAddEnumerable(ServiceDescriptor.Singleton<IConfigureOptions<JsonOptions>, ProblemDetailsJsonOptionsSetup>());
```

Comment:
Nit: Now that it's normal configure, it's more common just to combine it with the user-specified callback if any. something like.

```csharp
services.Configure<ProblemDetailsOptions>(options =>
{
    options.SerializerOptions.TypeInfoResolver =
        JsonTypeInfoResolver.Combine(AppJsonSerializerContext.Default, options.SerializerOptions.TypeInfoResolver);

    configure?.Invoke(options);
});
```

Reply:
That breaks this test that is explicitly testing that duplicates aren't added:

https://github.com/dotnet/aspnetcore/blob/138388372bb7c6e35fdfa679da740fabc275a688/src/Http/Http.Extensions/test/ProblemDetailsServiceCollectionExtensionsTest.cs#L33-L47

Reply:
And I was conflating two different options types lol.